### ISSUE TITLE
New authors don't get indexed properly for search

### ISSUE THREAD

**Issue by tfmorris** [internetarchive/openlibrary#2275]
### Description
It's a relatively common occurrence to come across OpenLibrary search listings which contain authors which have 0 works listed even though they have at least one work listed when you click through to their author record.

It occurred to me today that this is likely because the author entries are created by searching the Solr index, which likely hasn't yet been populated if both the work and author are getting added at the same time (as is likely with a new author's first book).

### Relevant url?
https://openlibrary.org/search/authors.json?q=Brandon+Hobson&mode=everything
http://openlibrary.org/authors/OL7444218A.json

### Expectation
All authors, including newly added authors, are indexed correctly.

### Proposal
We can do one of two things:
1. Create the new author doc for Solr directly using a placeholder template of:
```json
    {"work_count": 1,
    "top_work": "<new work title>",
    "top_subjects": "<subjects from new work>",
    }
```
2. Split the update into two and make sure that the update for the work is committed first.

I think I prefer the first approach.

**Comment by LeadSongDog** [comment]
Yes, timing is a definite factor.  After creating the new work and author records, further works cannot be linked to that author record until it has been indexed, which may take many minutes. A new contributor unfamiliar with the behavior will think that the author record was never created, often resulting in duplication.

**Comment by LeadSongDog** [comment]
Here's an older example. Two records that ImportBot created within 7 seconds back in 2008 for an author who already had a record:
{"name": "Grieg Marshall Spankie", "last_modified": {"type": "/type/datetime", "value": "**2008-04-30 09:38:13.731961**"}, "key": "/authors/OL3374795A", "type": {"key": "/type/author"}, "id": 13517871, "revision": 1}
{"name": "Grieg M. Spankie", "personal_name": "Grieg M. Spankie", "last_modified": {"type": "/type/datetime", "value": "**2008-09-16 18:45:08.660598**"}, "key": "/authors/OL4754134A", "type": {"key": "/type/author"}, "id": 20497658, "revision": 1}
{"name": "Grieg M. Spankie", "personal_name": "Grieg M. Spankie", "last_modified": {"type": "/type/datetime", "value": "**2008-09-16 18:45:15.392316**"}, "key": "/authors/OL4754143A", "type": {"key": "/type/author"}, "id": 20497692, "revision": 1}

**Comment by guyjeangilles** [comment]
Making this a sub-task of #789

**Comment by RayBB** [comment]
Echoing Drini's comment [here](https://github.com/internetarchive/openlibrary/issues/789#issuecomment-1819387729)
Solr has changed a lot in the past 6 years. I remember experiencing this issue long ago but haven't seen it in a while.

If anyone feels it's still an issue please create a new issue with updated examples and info.
